### PR TITLE
#37 refactored BoundBuffer

### PR DESCRIPTION
**Pull request by krzyk**
#37 
* refactored BoundBuffer to use byte array
* added tests to cover differences in implementation

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#52](https://github.com/yegor256/cactoos-http/issues/52) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#52](https://github.com/yegor256/cactoos-http/issues/52) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C63314D6Z/1361065218); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/test/java/org/cactoos/http/io/BoundedByteBufferTest.java [hunk 1]
```diff
@@ -21,35 +21,60 @@
...
         final int limit = 4;
         final BoundedByteBuffer buffer = new BoundedByteBuffer(limit);
-        final int[] bytes = new int[] {1, 2, 3, 4, 5, 6, 7, 8 };
+        final int[] bytes = {1, 2, 3, 4, 5, 6, 7, 8 };
         Arrays.stream(bytes).forEach(b -> buffer.offer((byte) b));
         MatcherAssert.assertThat(
             // @checkstyle MagicNumberCheck (1 line)
-            buffer.isEqualsTo(new byte[] {5, 6, 7, 8 }),
+            buffer.equalTo(new byte[] {5, 6, 7, 8 }),
             Matchers.is(true)
```
Comment (paulodamaso) [inline src/test/java/org/cactoos/http/io/BoundedByteBufferTest.java]: @krzyk How about getting rid of this `Matchers.is(true)` and use `new IsEqual<>(true)` like others test methods?

File: src/main/java/org/cactoos/http/io/BoundedByteBuffer.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     *
+     * @param bytes The bytes to compare to.
+     * @return The value {@code true} if the buffer contains exactly
+     *  the {@code bytes}.
+     */
+    public boolean equalTo(final byte[] bytes) {
+        final boolean result;
+        if (this.current < this.limit) {
+            result = Arrays.equals(
+                Arrays.copyOf(this.internal, this.current), bytes
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: @krzyk We should avoid using utility classes as `Arrays`; WDIT of using cactoos `Bytes` or `Equality` interfaces for this `byte[]` comparisons?

File: src/main/java/org/cactoos/http/io/BoundedByteBuffer.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            );
+        } else {
+            result = Arrays.equals(this.internal, bytes);
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: @krzyk Same as above, how about using cactoos `Bytes` or `Equality`.

**Comment by krzyk** [comment]
@paulodamaso updated, please take a look

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by krzyk** [comment]
@llorllale fixed build (Skipped class disappeared in 0.31 cactoos), please merge again

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#52](https://github.com/yegor256/cactoos-http/issues/52) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**